Okta.Xamarin/Okta.Xamarin/ViewModels/DiagnosticsViewModel.cs

```diff
@@ -3,36 +3,81 @@
 // Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
 // </copyright>
 
-using Okta.Xamarin.Views;
 using System;
+using System.Collections.Generic;
 using System.Security.Claims;
+using Okta.Xamarin.Views;
 using Xamarin.Forms;
 
 namespace Okta.Xamarin.ViewModels
 {
+    /// <summary>
+    /// View model for Diagnostics page.
+    /// </summary>
     public class DiagnosticsViewModel : BaseViewModel
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DiagnosticsViewModel"/> class.
+        /// </summary>
+        /// <param name="diagnosticsPage">The page.</param>
         public DiagnosticsViewModel(DiagnosticsPage diagnosticsPage)
         {
             this.Page = diagnosticsPage;
             this.StateManager = OktaContext.Current.StateManager;
+
+            // TODO: refactor this so handlers are only attached once. OKTA-373403
             OktaContext.Current.GetUserCompleted += (sender, args) =>
             {
-				this.userInfo = args.UserInfo;
-				this.ClaimsPrincipal = args.UserInfo as ClaimsPrincipal; // user info may be Dictionary<string, object> or ClaimsPrincipal or generic type; will be null if incorrect 'as' cast
-			};
+                this.UserInfo = args.UserInfo;
+                this.ClaimsPrincipal = args.UserInfo as ClaimsPrincipal; // user info may be Dictionary<string, object> or ClaimsPrincipal or generic type; will be null if incorrect 'as' cast
+            };
+            OktaContext.Current.IntrospectCompleted += (sender, args) =>
+            {
+                this.Page.DisplayData(args.Response as Dictionary<string, object>, "Introspection");
+            };
+            OktaContext.Current.RenewCompleted += (sender, args) =>
+            {
+                this.Page.DisplayData(args.Response.ToDictionary(), "Renewal");
+            };
```

Comment:
Just thoughts:
What about introducing functions in the View for populating Introspection and Renewal responses and calling them then from here like:
 `this.Page.DisplayIntrospectionData(args.Response as Dictionary<string, object>)`
...
`this.Page.DisplayRenewalData(args.Response.ToDictionary());`

and those functions will be calling `.DisplayData` extension method if needed. So private sections names will not be mentioned outside the view.

Reply:
I have additional items coming in v1.1 that will address that concern.